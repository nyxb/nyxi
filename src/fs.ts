import { resolve } from 'node:path'
import fs from 'node:fs'
import { consolji } from 'consolji'

export function getPackageJSON(cwd = process.cwd()): any {
  const path = resolve(cwd, 'package.json')

  if (fs.existsSync(path)) {
    try {
      const raw = fs.readFileSync(path, 'utf-8')
      const data = JSON.parse(raw)
      return data
    }
    catch (e) {
      consolji.warn('Failed to parse package.json')
      process.exit(0)
    }
  }
}
