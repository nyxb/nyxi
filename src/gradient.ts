import chalk from 'chalk'
import ora from 'ora'
import gradient from 'gradient-string'

const GREEN = '#14F195'
const PURPLE = '#9945FF'
const RED = '#ed4245'
const YELLOW = '#fee75c'

export const nyxGradient = gradient(GREEN, PURPLE)
export const nyxGreen = chalk.hex(GREEN)
export const nyxPurple = chalk.hex(PURPLE)
export const nyxRed = chalk.hex(RED)
export const yellow = chalk.hex(YELLOW)

export function nyxLoader(text: string) {
  return ora({
    text,
    spinner: {
      frames: ['   ', nyxGreen('>  '), nyxGreen('>> '), nyxGreen('>>>')],
    },
  })
}

export function info(...args: any[]) {
  // eslint-disable-next-line no-console
  console.log(nyxGreen.bold('>>>'), ...args)
}

export function error(...args: any[]) {
  console.error(nyxRed.bold('>>>'), ...args)
}

export function warn(...args: any[]) {
  console.error(yellow.bold('>>>'), ...args)
}
