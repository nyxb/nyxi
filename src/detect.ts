import fs from 'node:fs'
import path from 'node:path'
import { execaCommand } from 'execa'
import { findUp } from 'find-up'
import terminalLink from 'terminal-link'
import * as tyck from '@tyck/prompts'
import { consolji } from 'consolji'
import type { Agent } from './agents'
import { AGENTS, INSTALL_PAGE, LOCKS } from './agents'
import { cmdExists } from './utils'
import { getDefaultAgent } from './config'

export interface DetectOptions {
  autoInstall?: boolean
  cwd?: string
  skipPrompt?: boolean
}

export async function detect({ autoInstall, cwd, skipPrompt }: DetectOptions = {}) {
  let agent: Agent | null = null
  let version: string | null = null
  const lockPath = await findUp(Object.keys(LOCKS), { cwd })
  let packageJsonPath: string | undefined

  if (lockPath)
    packageJsonPath = path.resolve(lockPath, '../package.json')
  else
    packageJsonPath = await findUp('package.json', { cwd })

  // read `packageManager` field in package.json
  if (packageJsonPath && fs.existsSync(packageJsonPath)) {
    try {
      const pkg = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'))
      if (typeof pkg.packageManager === 'string') {
        const [name, ver] = pkg.packageManager.split('@')
        version = ver
        if (name === 'yarn' && parseInt(ver) > 1)
          agent = 'yarn@berry'
        else if (name === 'pnpm' && parseInt(ver) < 7)
          agent = 'pnpm@6'
        else if (name in AGENTS)
          agent = name
        else
          consolji.warn('[nyxi] Unknown packageManager:', pkg.packageManager)
      }
    }
    catch {}
  }

  // detect based on lock
  if (!agent && lockPath)
    agent = LOCKS[path.basename(lockPath)]

  // If no package manager is detected, prompt the user to choose one
  if (!agent && skipPrompt !== true) {
    const defaultAgent = await getDefaultAgent()
    if ((defaultAgent as any) === 'prompt') {
      agent = await tyck.select({
        message: 'Please choose a package manager:',
        options: [
          { value: 'pnpm', label: 'pnpm (recommended)' },
          { value: 'npm', label: 'npm' },
          { value: 'yarn', label: 'yarn' },
          { value: 'bun', label: 'bun' },
        ],
      }) as Agent
    }
    else {
      agent = defaultAgent as Agent
    }
  }
  // auto install
  if (agent && !cmdExists(agent.split('@')[0])) {
    if (!autoInstall) {
      consolji.warn(`[nyxi] Detected ${agent} but it doesn't seem to be installed.\n`)

      if (process.env.CI)
        process.exit(1)

      const link = terminalLink(agent, INSTALL_PAGE[agent])
      const tryInstall = await tyck.select({
        message: `Would you like to globally install ${link}?`,
        options: [
          { value: true, label: 'Yes' },
          { value: false, label: 'No' },
        ],
      })
      if (tryInstall === 0) {
        tyck.cancel('nevermind')
        process.exit(0)
      }
      else if (!tryInstall) {
        process.exit(1)
      }
    }

    await execaCommand(`npm i -g ${agent}${version ? `@${version}` : ''}`, { stdio: 'inherit', cwd })
  }

  return agent
}
