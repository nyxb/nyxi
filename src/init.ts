import fs from 'node:fs'
import { exec } from 'node:child_process'
import consolji from 'consolji'
import { multiselect, select } from '@tyck/prompts'

export async function createConfigFiles() {
  async function initPackageJson() {
    if (fs.existsSync('package.json')) {
      consolji.log('⚠️ package.json already exists.')
    }
    else {
      const packageJson = {
        name: 'example',
        version: '0.0.1',
        description: '',
        author: '',
        license: 'MIT',
        hompage: 'https://💻nyxb.ws',
        repository: {
          type: 'git',
          url: 'git+https://example.com.git',
        },
        bugs: {
          url: 'https://github.com/example/issues',
        },
        bin: {
          example: '/dist/example.js',
        },
        main: 'example.js',
        module: 'example.js',
        types: 'example.d.ts',
        exports: {
          '.': {
            require: './dist/example.js',
            import: './dist/example.mjs',
            types: './dist/example.d.ts',
          },
        },
        scripts: {
          test: 'echo "Happy hacking" && 💙',
        },
        dependencies: {},
        devDependencies: {},
        files: [],
        keywords: [],
      }

      fs.writeFileSync('package.json', JSON.stringify(packageJson, null, 2))
      consolji.log('🎉 package.json created.')
    }
  }

  async function initTsConfig() {
    if (fs.existsSync('tsconfig.json')) {
      consolji.log('⚠️ tsconfig.json already exists.')
    }
    else {
      const tsconfig = {
        compilerOptions: {
          target: 'ESNext',
          module: 'ESNext',
          moduleResolution: 'Node',
          strict: true,
          esModuleInterop: true,
          skipLibCheck: true,
          forceConsistentCasingInFileNames: true,
        },
        include: ['src'],
      }

      fs.writeFileSync('tsconfig.json', JSON.stringify(tsconfig, null, 2))
      consolji.log('🎉 tsconfig.json created.')
    }
  }

  const createConfigs = await multiselect({
    message: 'Which files do you want to create?',
    options: [
      { value: 'package.json', label: 'Create package.json' },
      { value: 'tsconfig.json', label: 'Create tsconfig.json' },
    ],
  }) as string[]

  if (createConfigs.includes('package.json'))
    await initPackageJson()

  if (createConfigs.includes('tsconfig.json'))
    await initTsConfig()

  const openWithVSCode = await select({
    message: 'Do you want to open the files with Visual Studio Code?',
    options: [
      { value: 'yes', label: 'Yes' },
      { value: 'no', label: 'No' },
    ],
  })

  if (openWithVSCode === 'yes') {
    exec('code .', (error, stdout, stderr) => {
      if (error) {
        consolji.error(`exec error: ${error}`)
        return
      }
      consolji.log(`stdout: ${stdout}`)
      consolji.error(`stderr: ${stderr}`)
    })
  }

  return createConfigs
}
