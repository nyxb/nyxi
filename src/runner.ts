import { resolve } from 'node:path'
import * as tyck from '@tyck/prompts'
import { execaCommand } from 'execa'
import color from '@nyxb/picocolors'
import { consolji } from 'consolji'
import { version } from '../package.json'
import type { Agent } from './agents'
import { agents } from './agents'
import { getDefaultAgent, getGlobalAgent } from './config'
import type { DetectOptions } from './detect'
import { detect } from './detect'
import { getVoltaPrefix, remove } from './utils'
import { UnsupportedCommand } from './parse'

const DEBUG_SIGN = '?'

export interface RunnerContext {
  hasLock?: boolean
  cwd?: string
}

export type Runner = (agent: Agent, args: string[], ctx?: RunnerContext) => Promise<string | undefined> | string | undefined

export async function runCli(fn: Runner, options: DetectOptions = {}, skipPrompt = false) {
  const args = process.argv.slice(2).filter(Boolean)
  try {
    await run(fn, args, options, skipPrompt)
  }
  catch (err: any) {
    if (err instanceof UnsupportedCommand)
      consolji.error(color.red(` ${err.message}`))

    process.exit(1)
  }
}

export async function run(fn: Runner, args: string[], options: DetectOptions = {}, skipPrompt = false) {
  const debug = args.includes(DEBUG_SIGN)
  if (debug)
    remove(args, DEBUG_SIGN)

  let cwd = process.cwd()
  let command

  if (args.length === 1 && (args[0] === '--version' || args[0] === '-v')) {
    consolji.info(`${color.nicegreen('@nyxb/nyxi')} v${color.yellow(`${version}`)}`)
    return
  }

  if (args.length === 1 && ['-h', '--help'].includes(args[0])) {
    const dash = color.dim('-')
    consolji.log(color.green(color.bold('@nyxb/nyxi')) + color.dim(` use the right package manager v${version}\n`))
    consolji.info(`nyxi   ${dash}  install`)
    consolji.info(`nyxr   ${dash}  run`)
    consolji.info(`nyxlx  ${dash}  execute`)
    consolji.info(`nyxu   ${dash}  upgrade`)
    consolji.info(`nyxun  ${dash}  uninstall`)
    consolji.info(`nyxci  ${dash}  clean install`)
    consolji.info(`nyxa   ${dash}  agent alias`)
    consolji.info(`${color.yellow('check')} ${color.purple('https://github.com/nyxb/nyxi')} for more documentation.`)
    return
  }

  if (args[0] === '-C') {
    cwd = resolve(cwd, args[1])
    args.splice(0, 2)
  }

  const isGlobal = args.includes('-g')
  if (isGlobal) {
    command = await fn(await getGlobalAgent(), args)
  }
  else {
    let agent: string = await detect({ ...options, cwd }) || await getDefaultAgent()
    if (!agents.includes(agent as Agent) && !skipPrompt) {
      const result = await tyck.select({
        message: 'Choose the agent',
        options: agents.filter(i => !i.includes('@')).map(value => ({ label: value, value })),
      })

      if (tyck.isCancel(result)) {
        tyck.cancel('nevermind')
        process.exit(0)
      }
      agent = result as Agent

      if (!agent)
        return
    }

    command = await fn(agent as Agent, args, {
      hasLock: Boolean(agent),
      cwd,
    })
  }

  if (!command)
    return

  const voltaPrefix = getVoltaPrefix()
  if (voltaPrefix)
    command = voltaPrefix.concat(' ').concat(command)

  if (debug) {
    consolji.log(command)
    return
  }

  await execaCommand(command, { stdio: 'inherit', encoding: 'utf-8', cwd })
}
