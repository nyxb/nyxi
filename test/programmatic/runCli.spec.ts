import path from 'node:path'
import { tmpdir } from 'node:os'
import fs from 'fs-extra'
import type { SpyInstance } from 'vitest'
import { afterAll, beforeAll, describe, expect, it, vi } from 'vitest'
import { AGENTS, parseNyxa, parseNyxi, parseNyxlx, parseNyxu, parseNyxun, runCli } from '../../src'

import type { Runner } from '../../src'

let basicLog: SpyInstance, errorLog: SpyInstance, warnLog: SpyInstance, infoLog: SpyInstance

function runCliTest(fixtureName: string, agent: string, runner: Runner, args: string[]) {
   return async () => {
      const cwd = await fs.mkdtemp(path.join(tmpdir(), 'nyxi-'))
      const fixture = path.join(__dirname, '..', 'fixtures', fixtureName, agent)
      await fs.copy(fixture, cwd)

      await runCli(
         async (agent, _, ctx) => {
            // we override the args to be test specific
            return runner(agent, args, ctx)
         },
         {
            programmatic: true,
            cwd,
            args,
         },
      ).catch((e) => {
      // it will always throw if execa is mocked
         if (e.command)
            expect(e.command).toMatchSnapshot()
         else
            expect(e.message).toMatchSnapshot()
      })
   }
}

beforeAll(() => {
   basicLog = vi.spyOn(console, 'log')
   warnLog = vi.spyOn(console, 'warn')
   errorLog = vi.spyOn(console, 'error')
   infoLog = vi.spyOn(console, 'info')

   vi.mock('execa', async (importOriginal) => {
      const mod = await importOriginal() as any
      return {
         ...mod,
         execaCommand: (cmd: string) => {
            // break execution flow for easier snapshotting
            // eslint-disable-next-line no-throw-literal
            throw { command: cmd }
         },
      }
   })
})

afterAll(() => {
   vi.resetAllMocks()
})

const agents = [...Object.keys(AGENTS), 'unknown']
const fixtures = ['lockfile', 'packager']

// matrix testing of: fixtures x agents x commands
fixtures.forEach(fixture => describe(fixture, () => agents.forEach(agent => describe(agent, () => {
   /** nyxa */
   it('nyxa', runCliTest(fixture, agent, parseNyxa, []))
   it('nyxa run foo', runCliTest(fixture, agent, parseNyxa, ['run', 'foo']))

   /** nyxi */
   it('nyxi', runCliTest(fixture, agent, parseNyxi, []))
   it('nyxi foo', runCliTest(fixture, agent, parseNyxi, ['foo']))
   it('nyxi foo -D', runCliTest(fixture, agent, parseNyxi, ['foo', '-D']))
   it('nyxi --frozen', runCliTest(fixture, agent, parseNyxi, ['--frozen']))
   it('nyxi -g foo', runCliTest(fixture, agent, parseNyxi, ['-g', 'foo']))

   /** nyxlx */
   it('nyxlx', runCliTest(fixture, agent, parseNyxlx, ['foo']))

   /** nyxu */
   it('nyxu', runCliTest(fixture, agent, parseNyxu, []))
   it('nyxu -i', runCliTest(fixture, agent, parseNyxu, ['-i']))

   /** nyxun */
   it('nyxun foo', runCliTest(fixture, agent, parseNyxun, ['foo']))
   it('nyxun -g foo', runCliTest(fixture, agent, parseNyxun, ['-g', 'foo']))

   it('no logs', () => {
      expect(basicLog).not.toHaveBeenCalled()
      expect(warnLog).not.toHaveBeenCalled()
      expect(errorLog).not.toHaveBeenCalled()
      expect(infoLog).not.toHaveBeenCalled()
   })
}))))
