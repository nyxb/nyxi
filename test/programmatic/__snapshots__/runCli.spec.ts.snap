// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lockfile > bun > nyxa 1`] = `"bun"`;

exports[`lockfile > bun > nyxa run foo 1`] = `"bun run foo"`;

exports[`lockfile > bun > nyxi --frozen 1`] = `"bun install --no-save"`;

exports[`lockfile > bun > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > bun > nyxi 1`] = `"bun install"`;

exports[`lockfile > bun > nyxi foo -D 1`] = `"bun add foo -d"`;

exports[`lockfile > bun > nyxi foo 1`] = `"bun add foo"`;

exports[`lockfile > bun > nyxlx 1`] = `"bunx foo"`;

exports[`lockfile > bun > nyxu -i 1`] = `"bun update"`;

exports[`lockfile > bun > nyxu 1`] = `"bun update"`;

exports[`lockfile > bun > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > bun > nyxun foo 1`] = `"bun remove foo"`;

exports[`lockfile > npm > nyxa 1`] = `"npm"`;

exports[`lockfile > npm > nyxa run foo 1`] = `"npm run foo"`;

exports[`lockfile > npm > nyxi --frozen 1`] = `"npm ci"`;

exports[`lockfile > npm > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > npm > nyxi 1`] = `"npm i"`;

exports[`lockfile > npm > nyxi foo -D 1`] = `"npm i foo -D"`;

exports[`lockfile > npm > nyxi foo 1`] = `"npm i foo"`;

exports[`lockfile > npm > nyxlx 1`] = `"npx foo"`;

exports[`lockfile > npm > nyxu -i 1`] = `"Command \\"upgrade-interactive\\" is not support by agent \\"npm\\""`;

exports[`lockfile > npm > nyxu 1`] = `"npm update"`;

exports[`lockfile > npm > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > npm > nyxun foo 1`] = `"npm uninstall foo"`;

exports[`lockfile > pnpm > nyxa 1`] = `"pnpm"`;

exports[`lockfile > pnpm > nyxa run foo 1`] = `"pnpm run foo"`;

exports[`lockfile > pnpm > nyxi --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`lockfile > pnpm > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > pnpm > nyxi 1`] = `"pnpm i"`;

exports[`lockfile > pnpm > nyxi foo -D 1`] = `"pnpm add foo -D"`;

exports[`lockfile > pnpm > nyxi foo 1`] = `"pnpm add foo"`;

exports[`lockfile > pnpm > nyxlx 1`] = `"pnpm dlx foo"`;

exports[`lockfile > pnpm > nyxu -i 1`] = `"pnpm update -i"`;

exports[`lockfile > pnpm > nyxu 1`] = `"pnpm update"`;

exports[`lockfile > pnpm > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > pnpm > nyxun foo 1`] = `"pnpm remove foo"`;

exports[`lockfile > pnpm@6 > nyxa 1`] = `"pnpm"`;

exports[`lockfile > pnpm@6 > nyxa run foo 1`] = `"pnpm run foo"`;

exports[`lockfile > pnpm@6 > nyxi --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`lockfile > pnpm@6 > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > pnpm@6 > nyxi 1`] = `"pnpm i"`;

exports[`lockfile > pnpm@6 > nyxi foo -D 1`] = `"pnpm add foo -D"`;

exports[`lockfile > pnpm@6 > nyxi foo 1`] = `"pnpm add foo"`;

exports[`lockfile > pnpm@6 > nyxlx 1`] = `"pnpm dlx foo"`;

exports[`lockfile > pnpm@6 > nyxu -i 1`] = `"pnpm update -i"`;

exports[`lockfile > pnpm@6 > nyxu 1`] = `"pnpm update"`;

exports[`lockfile > pnpm@6 > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > pnpm@6 > nyxun foo 1`] = `"pnpm remove foo"`;

exports[`lockfile > unknown > nyxa 1`] = `"pnpm"`;

exports[`lockfile > unknown > nyxa run foo 1`] = `"pnpm run foo"`;

exports[`lockfile > unknown > nyxi --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`lockfile > unknown > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > unknown > nyxi 1`] = `"pnpm i"`;

exports[`lockfile > unknown > nyxi foo -D 1`] = `"pnpm add foo -D"`;

exports[`lockfile > unknown > nyxi foo 1`] = `"pnpm add foo"`;

exports[`lockfile > unknown > nyxlx 1`] = `"pnpm dlx foo"`;

exports[`lockfile > unknown > nyxu -i 1`] = `"pnpm update -i"`;

exports[`lockfile > unknown > nyxu 1`] = `"pnpm update"`;

exports[`lockfile > unknown > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > unknown > nyxun foo 1`] = `"pnpm remove foo"`;

exports[`lockfile > yarn > nyxa 1`] = `"yarn"`;

exports[`lockfile > yarn > nyxa run foo 1`] = `"yarn run foo"`;

exports[`lockfile > yarn > nyxi --frozen 1`] = `"yarn install --frozen-lockfile"`;

exports[`lockfile > yarn > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > yarn > nyxi 1`] = `"yarn install"`;

exports[`lockfile > yarn > nyxi foo -D 1`] = `"yarn add foo -D"`;

exports[`lockfile > yarn > nyxi foo 1`] = `"yarn add foo"`;

exports[`lockfile > yarn > nyxlx 1`] = `"npx foo"`;

exports[`lockfile > yarn > nyxu -i 1`] = `"yarn upgrade-interactive"`;

exports[`lockfile > yarn > nyxu 1`] = `"yarn upgrade"`;

exports[`lockfile > yarn > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > yarn > nyxun foo 1`] = `"yarn remove foo"`;

exports[`lockfile > yarn@berry > nyxa 1`] = `"yarn"`;

exports[`lockfile > yarn@berry > nyxa run foo 1`] = `"yarn run foo"`;

exports[`lockfile > yarn@berry > nyxi --frozen 1`] = `"yarn install --frozen-lockfile"`;

exports[`lockfile > yarn@berry > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`lockfile > yarn@berry > nyxi 1`] = `"yarn install"`;

exports[`lockfile > yarn@berry > nyxi foo -D 1`] = `"yarn add foo -D"`;

exports[`lockfile > yarn@berry > nyxi foo 1`] = `"yarn add foo"`;

exports[`lockfile > yarn@berry > nyxlx 1`] = `"npx foo"`;

exports[`lockfile > yarn@berry > nyxu -i 1`] = `"yarn upgrade-interactive"`;

exports[`lockfile > yarn@berry > nyxu 1`] = `"yarn upgrade"`;

exports[`lockfile > yarn@berry > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`lockfile > yarn@berry > nyxun foo 1`] = `"yarn remove foo"`;

exports[`packager > bun > nyxa 1`] = `"bun"`;

exports[`packager > bun > nyxa run foo 1`] = `"bun run foo"`;

exports[`packager > bun > nyxi --frozen 1`] = `"bun install --no-save"`;

exports[`packager > bun > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`packager > bun > nyxi 1`] = `"bun install"`;

exports[`packager > bun > nyxi foo -D 1`] = `"bun add foo -d"`;

exports[`packager > bun > nyxi foo 1`] = `"bun add foo"`;

exports[`packager > bun > nyxlx 1`] = `"bunx foo"`;

exports[`packager > bun > nyxu -i 1`] = `"bun update"`;

exports[`packager > bun > nyxu 1`] = `"bun update"`;

exports[`packager > bun > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > bun > nyxun foo 1`] = `"bun remove foo"`;

exports[`packager > npm > nyxa 1`] = `"npm"`;

exports[`packager > npm > nyxa run foo 1`] = `"npm run foo"`;

exports[`packager > npm > nyxi --frozen 1`] = `"npm ci"`;

exports[`packager > npm > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`packager > npm > nyxi 1`] = `"npm i"`;

exports[`packager > npm > nyxi foo -D 1`] = `"npm i foo -D"`;

exports[`packager > npm > nyxi foo 1`] = `"npm i foo"`;

exports[`packager > npm > nyxlx 1`] = `"npx foo"`;

exports[`packager > npm > nyxu -i 1`] = `"Command \\"upgrade-interactive\\" is not support by agent \\"npm\\""`;

exports[`packager > npm > nyxu 1`] = `"npm update"`;

exports[`packager > npm > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > npm > nyxun foo 1`] = `"npm uninstall foo"`;

exports[`packager > pnpm > nyxa 1`] = `"pnpm"`;

exports[`packager > pnpm > nyxa run foo 1`] = `"pnpm run foo"`;

exports[`packager > pnpm > nyxi --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > pnpm > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`packager > pnpm > nyxi 1`] = `"pnpm i"`;

exports[`packager > pnpm > nyxi foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > pnpm > nyxi foo 1`] = `"pnpm add foo"`;

exports[`packager > pnpm > nyxlx 1`] = `"pnpm dlx foo"`;

exports[`packager > pnpm > nyxu -i 1`] = `"pnpm update -i"`;

exports[`packager > pnpm > nyxu 1`] = `"pnpm update"`;

exports[`packager > pnpm > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > pnpm > nyxun foo 1`] = `"pnpm remove foo"`;

exports[`packager > pnpm@6 > nyxa 1`] = `"pnpm"`;

exports[`packager > pnpm@6 > nyxa run foo 1`] = `"pnpm run foo"`;

exports[`packager > pnpm@6 > nyxi --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > pnpm@6 > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`packager > pnpm@6 > nyxi 1`] = `"pnpm i"`;

exports[`packager > pnpm@6 > nyxi foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > pnpm@6 > nyxi foo 1`] = `"pnpm add foo"`;

exports[`packager > pnpm@6 > nyxlx 1`] = `"pnpm dlx foo"`;

exports[`packager > pnpm@6 > nyxu -i 1`] = `"pnpm update -i"`;

exports[`packager > pnpm@6 > nyxu 1`] = `"pnpm update"`;

exports[`packager > pnpm@6 > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > pnpm@6 > nyxun foo 1`] = `"pnpm remove foo"`;

exports[`packager > unknown > nyxa 1`] = `"pnpm"`;

exports[`packager > unknown > nyxa run foo 1`] = `"pnpm run foo"`;

exports[`packager > unknown > nyxi --frozen 1`] = `"pnpm i --frozen-lockfile"`;

exports[`packager > unknown > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`packager > unknown > nyxi 1`] = `"pnpm i"`;

exports[`packager > unknown > nyxi foo -D 1`] = `"pnpm add foo -D"`;

exports[`packager > unknown > nyxi foo 1`] = `"pnpm add foo"`;

exports[`packager > unknown > nyxlx 1`] = `"pnpm dlx foo"`;

exports[`packager > unknown > nyxu -i 1`] = `"pnpm update -i"`;

exports[`packager > unknown > nyxu 1`] = `"pnpm update"`;

exports[`packager > unknown > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > unknown > nyxun foo 1`] = `"pnpm remove foo"`;

exports[`packager > yarn > nyxa 1`] = `"yarn"`;

exports[`packager > yarn > nyxa run foo 1`] = `"yarn run foo"`;

exports[`packager > yarn > nyxi --frozen 1`] = `"yarn install --frozen-lockfile"`;

exports[`packager > yarn > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`packager > yarn > nyxi 1`] = `"yarn install"`;

exports[`packager > yarn > nyxi foo -D 1`] = `"yarn add foo -D"`;

exports[`packager > yarn > nyxi foo 1`] = `"yarn add foo"`;

exports[`packager > yarn > nyxlx 1`] = `"npx foo"`;

exports[`packager > yarn > nyxu -i 1`] = `"yarn upgrade-interactive"`;

exports[`packager > yarn > nyxu 1`] = `"yarn upgrade"`;

exports[`packager > yarn > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > yarn > nyxun foo 1`] = `"yarn remove foo"`;

exports[`packager > yarn@berry > nyxa 1`] = `"yarn"`;

exports[`packager > yarn@berry > nyxa run foo 1`] = `"yarn run foo"`;

exports[`packager > yarn@berry > nyxi --frozen 1`] = `"yarn install --immutable"`;

exports[`packager > yarn@berry > nyxi -g foo 1`] = `"npm i -g foo"`;

exports[`packager > yarn@berry > nyxi 1`] = `"yarn install"`;

exports[`packager > yarn@berry > nyxi foo -D 1`] = `"yarn add foo -D"`;

exports[`packager > yarn@berry > nyxi foo 1`] = `"yarn add foo"`;

exports[`packager > yarn@berry > nyxlx 1`] = `"yarn dlx foo"`;

exports[`packager > yarn@berry > nyxu -i 1`] = `"yarn up -i"`;

exports[`packager > yarn@berry > nyxu 1`] = `"yarn up"`;

exports[`packager > yarn@berry > nyxun -g foo 1`] = `"npm uninstall -g foo"`;

exports[`packager > yarn@berry > nyxun foo 1`] = `"yarn remove foo"`;
